src/Raven.Server/ServerWide/Commands/AI/AddOrUpdateAiAgentCommand.cs

```diff
@@ -8,27 +8,25 @@ namespace Raven.Server.ServerWide.Commands.AI
 {
     public class AddOrUpdateAiAgentCommand : UpdateDatabaseCommand
     {
-        public string AgentName;
         public AiAgentConfiguration Configuration;
 
         public AddOrUpdateAiAgentCommand()
         {
             // for deserialization    
         }
 
-        public AddOrUpdateAiAgentCommand(string database, string agentName, AiAgentConfiguration configuration, string uniqueRequestId) : base(database, uniqueRequestId)
+        public AddOrUpdateAiAgentCommand(string database, AiAgentConfiguration configuration, string uniqueRequestId) : base(database, uniqueRequestId)
         {
             Configuration = configuration ?? throw new ArgumentNullException(nameof(configuration)); 
-            AgentName = agentName ?? throw new ArgumentNullException(nameof(agentName));
         }
         public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
-            record.AiAgents[AgentName] = Configuration;
+            record.AiAgents.RemoveAll(c => c.Identifier == Configuration.Identifier);
```

Comment:
we need to go through `EnsureTaskNameIsNotUsed` / `EnsureUniqueTaskName` and friends

Reply:
Done, although it isn't a `Task`